tests/test_rotations.py

```diff
@@ -0,0 +1,110 @@
+import numpy as np
+import pytest
+from numpy.typing import NDArray
+from scipy.spatial.transform import Rotation
+
+from timemachine.lib import custom_ops
+
+
+def hamiliton_product(q1: NDArray, q2: NDArray) -> NDArray:
+    """
+    Compute the product of two quaternions, called a hamilton product.
+
+    References
+    ----------
+    https://en.wikipedia.org/wiki/Quaternion#Hamilton_product
+    """
+    w1, x1, y1, z1 = q1[0], q1[1], q1[2], q1[3]
+    w2, x2, y2, z2 = q2[0], q2[1], q2[2], q2[3]
+    output = np.empty(4)
+    output[0] = (w1 * w2) - (x1 * x2) - (y1 * y2) - (z1 * z2)
+    output[1] = (w1 * x2) + (x1 * w2) + (y1 * z2) - (z1 * y2)
+    output[2] = (w1 * y2) - (x1 * z2) + (y1 * w2) + (z1 * x2)
+    output[3] = (w1 * z2) + (x1 * y2) - (y1 * x2) + (z1 * w2)
+    return output
+
+
+def rotate_vector_by_quaternions(coords: NDArray, quaternion: NDArray) -> NDArray:
```

Comment:
Not sure if we want this reference implmentation outside the test file, since you can use scipy.Rotation instead, and this isn't intended to be performant.

Reply:
Any reason we couldn't just use the scipy implementation as the reference and remove this entirely?

Reply:
Totally could, had done this primarily to have a 1-1 implementation of the quaternion rotation rather than going through the matrix multiplication that Scipy does (https://github.com/scipy/scipy/blob/main/scipy/spatial/transform/_rotation.pyx#L1943).

Reply:
Removed the ref implementation here:  https://github.com/proteneer/timemachine/pull/1174/commits/251899e6e8f72a1e4031d4291dad99f219526d3a